### PR TITLE
Fixed disable/enable logic in PassConfig

### PR DESCRIPTION
**Pull request by GlebKazantaev**
In this PR I fixed PassConfig logic for cases when passes are disabled by default but enabled externally via PassConfig. Because of variety of use-cases this one weren't covered before. All problematic cases I addressed in unit tests.

### CODE REVIEW

File: ngraph/core/include/ngraph/pass/manager.hpp [hunk 1]
```diff
@@ -102,6 +102,7 @@ class NGRAPH_API ngraph::pass::Manager
     /// \brief Set external PassConfig object.
     void set_pass_config(const std::shared_ptr<PassConfig>& pass_config)
     {
+        pass_config->add_disabled_passes(*m_pass_config);
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/pass/manager.hpp:105]: From my point of view it is unexpected that we disabled some passes in the constant argument in method which has a name `set_pass_config`

File: ngraph/core/src/pass/graph_rewrite.cpp [hunk 2]
```diff
@@ -272,6 +272,19 @@ void pass::GraphRewrite::add_matcher(const shared_ptr<pattern::Matcher>& m,
     NGRAPH_SUPPRESS_DEPRECATED_END
 }
 
+void pass::GraphRewrite::set_pass_config(const std::shared_ptr<PassConfig>& rhs)
+{
+    auto pass_config = get_pass_config();
+    rhs->add_disabled_passes(*pass_config);
```
Comment (ilyachur) [inline ngraph/core/src/pass/graph_rewrite.cpp:278]: The same comment as above.

File: ngraph/core/include/ngraph/pass/manager.hpp (hunk 1 above)
Comment (GlebKazantaev) [reply ngraph/core/include/ngraph/pass/manager.hpp:105]: During offline discussion decided to pass shared pass config to Manager constructor and remove set_pass_config method for pass::Manager.

File: ngraph/core/src/pass/graph_rewrite.cpp (hunk 2 above)
Comment (GlebKazantaev) [reply ngraph/core/src/pass/graph_rewrite.cpp:278]: During offline discussion decided to keep it as is but add some explanation comments.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**